scripts/fauxalysis_from_docking.py

```diff
@@ -1,39 +1,242 @@
+        Top-level directory of original-style output
+    out_dir : str
+        Top-level directory of new output
+    frag_dir : str
+        Path to fragalysis results directory.
+    best_structure_dict : dict
+        Of the form {('Compound_ID', dimer): 'Structure_Source'} where "dimer" is a boolean.
+    """
+    os.makedirs(f"{out_dir}", exist_ok=True)
+
+    ## Create list of sdf files we will make so that we can concatenate them at the end
+    cmpd_sdf_list = []
+
+    ## Loop through dict and parse input files into output files
+    for (compound_id, dimer), best_str in best_structure_dict.items():
+        ## Make sure input exists
+        dimer_s = "dimer" if dimer else "monomer"
+        compound_in_dir = f"{in_dir}/{compound_id}/{dimer_s}/{best_str}"
+        compound_out_dir = f"{out_dir}/{compound_id}"
+
+        ## If inputs don't exist, else if the output directory already exists, don't waste time
+        if not check_output(compound_in_dir):
+            print(
+                (
+                    f"No results found for {compound_id}/{dimer_s}/{best_str}, "
+                    "skipping"
+                ),
+                flush=True,
+            )
+            continue
+        else:
+            print(
+                (
+                    f"Generating fauxalysis...\n"
+                    f"\tfrom: {compound_in_dir}\n"
+                    f"\tto: {compound_out_dir}"
+                ),
+                flush=True,
+            )
+
+        ## Create output directory if necessary
+        os.makedirs(compound_out_dir, exist_ok=True)
+
+        ## Load necessary files
+        du = oechem.OEDesignUnit()
+        oechem.OEReadDesignUnit(f"{compound_in_dir}/predocked.oedu", du)
+        prot = oechem.OEGraphMol()
+        du.GetProtein(prot)
+        lig = load_openeye_sdf(f"{compound_in_dir}/docked.sdf")
+
+        ## Set ligand title, useful for the combined sdf file
+        lig.SetTitle(f"{compound_id}_{best_str}")
+
+        ## Give ligand atoms their own chain "L" and set the resname to "LIG"
+        residue = oechem.OEAtomGetResidue(next(iter(lig.GetAtoms())))
+        residue.SetChainID("L")
+        residue.SetName("LIG")
+        residue.SetHetAtom(True)
+        for atom in list(lig.GetAtoms()):
+            oechem.OEAtomSetResidue(atom, residue)
+
+        ## First save apo
+        save_openeye_pdb(prot, f"{compound_out_dir}/{best_str}_apo.pdb")
+
+        ## Combine protein and ligand and save
+        ## TODO: consider saving water as well
+        oechem.OEAddMols(prot, lig)
+
+        ## Clean up PDB info by re-perceiving, perserving chain ID, residue number, and residue name
+        preserve = (
+            oechem.OEPreserveResInfo_ChainID
+            | oechem.OEPreserveResInfo_ResidueNumber
+            | oechem.OEPreserveResInfo_ResidueName
+        )
+        oechem.OEPerceiveResidues(prot, preserve)
+
+        save_openeye_pdb(prot, f"{compound_out_dir}/{compound_id}_bound.pdb")
```

Comment:
This addresses some of these issues:
#75

Reply:
I think this code should be moved to its own API somewhere (maybe in openeye or in modeling).

Reply:
done in https://github.com/choderalab/covid-moonshot-ml/pull/65/commits/f9a8a8b8e5b8dc8c418d9971f86bd6fdc30ee2b0